### PR TITLE
Allow player_t to store and gain moves

### LINKED ISSUE #1053: Allow player_t to store and gain moves
Currently, there is no way for a player to unlock moves for future use in battle. To pave the way for future implementation of this functionality, we will add a linked list of `move`s to the `player_t` struct, and refactor the `add_move` function to take in a `player_t` instead of a `combatant_t`.

### PR DESCRIPTION
**Pull request by adinilfeld**
Currently, there is no way for a player to unlock moves for future use in battle. To pave the way for future implementation of this functionality, several changes have been made: the `player_t` struct now has a field to store a linked list of moves; the `add_move` function has been refactored to take in a `player_t` instead of a `combatant_t`; and `add_move` has been moved from battle to player. Lastly, the unit tests associated with `add_move` were moved from battle to player as well, and now pass successfully.

Closes #1053.

### CODE REVIEW

File: src/game-state/src/player.c [hunk 1]
```diff
@@ -267,4 +268,23 @@ int player_add_stat_effect(player_t *player, stat_effect_t *effect)
...
+
+    move_t *last_move = player->moves;
+    
+    if (player->moves == NULL){
+        player->moves = move;
+        return SUCCESS;
+    }
+
+    while (last_move->next != NULL) {
+        last_move = last_move->next;
```
Comment (WPDolan) [inline src/game-state/src/player.c:286]: Why are new moves added to the end of the list?

**Review by WPDolan — COMMENTED** [review]
Everything looks good! I just have one design question, but the new additions to the player struct are much appreciated!

File: src/game-state/src/player.c (hunk 1 above)
Comment (adinilfeld) [reply src/game-state/src/player.c:286]: That was the previous implementation when the function was in Battles; I think it also matches a lot of other video games where new moves get added to the end of some kind of moves menu. It wouldn't be too hard to change though if this isn't ideal.

Comment (WPDolan) [reply src/game-state/src/player.c:286]: That makes sense. Thank you for the clarification!

**Review by WPDolan — APPROVED** [review]
My design question has been answered, and the changes to the player module look good! Thank you for adding moves to `player_t`!

**Review by tgao564 — APPROVED** [review]
This all looks great to me! This will open up a lot of room for future integration with the moves struct. I've already created a backlog sprint issue for skill trees to start working with the new moves struct. Good work!

**Review by firatciftci — APPROVED** [review]
Everything looks good to me!

**Comment by adinilfeld** [comment]
To summarize this merge: several changes have been made to allow future SkillTrees teams to implement skills which unlock battle moves. Namely, the `player_t` struct now has a field to store a linked list of moves; the `add_move` function has been refactored to take in a `player_t` instead of a `combatant_t`; and `add_move` has been moved from battle to player. Lastly, the unit tests associated with `add_move` were moved from battle to player as well, and now pass successfully.

**Comment by firatciftci** [comment]
Issue Score: ✔️++

Comments:
Great work!

**Pull request merged**